src/internal/packager2/parse/load.go

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package parse parses zarf.yaml files into Zarf objects
+package parse
+
+import (
+	"context"
+
+	goyaml "github.com/goccy/go-yaml"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/pkg/logger"
+)
+
+// ZarfPackage parses the yaml passed as a byte slice and applies potential schema migrations.
+func ZarfPackage(ctx context.Context, b []byte) (v1alpha1.ZarfPackage, error) {
```

Comment:
Does this need its own package? ParseZarfPackage seems to be the function name, maybe it can be a private func in load?

Reply:
I originally had the same thought, but it can't be in load since both `load` and `packageLayout` use `parse.ZarfPackage`. It would create a new cyclic dependency. 
![Image](https://github.com/user-attachments/assets/598332ed-9d12-4e36-af3b-bebc4d433462)